pkg/cmd/util/route.go

```diff
@@ -54,10 +57,16 @@ func UnsecuredRoute(kc kclientset.Interface, namespace, routeName, serviceName,
 		},
 	}
 
-	// If the service has multiple ports and the user didn't specify --port,
-	// then default the route port to a service port name.
-	if len(port.Name) > 0 && len(portString) == 0 {
-		route.Spec.Port = resolveRoutePort(port.Name)
+	// When route.Spec.Port is not set, the generator will pick a service port.
+
+	// If the user didn't specify --port, and either the service has a port.Name
+	// or forcePort is set, set route.Spec.Port
+	if (len(port.Name) > 0 || forcePort) && len(portString) == 0 {
+		if forcePort {
+			route.Spec.Port = getTargetPort(svc)
+		} else {
+			route.Spec.Port = resolveRoutePort(port.Name)
```

Comment:
Do we need this function?  Can't we just conditionally call getTargetPort with svc.Spec.Ports[0].TargetPort?

And is there ever a condition where a service does not have a target port?  Did you check to make sure that was explicitly illegal in the type definition?  Or do we need to be tolerant of it?